src/sentry/search/events/fields.py

```diff
@@ -2547,6 +2599,149 @@ def _resolve_unimplemented_alias(self, alias: str) -> SelectType:
+        lcp_index = Function(
+            "indexOf", [self.column("measurements_key"), "lcp"], MEASUREMENTS_LCP_INDEX_ALIAS
+        )
+
+        return Function(
+            "multiIf",
+            [
+                Function(
+                    "equals",
+                    [
+                        Function(
+                            "tupleElement",
+                            [self.resolve_field("project_threshold_config"), 1],
+                        ),
+                        "lcp",
+                    ],
+                ),
+                Function(
+                    "if",
+                    [
+                        Function("equals", [lcp_index, 0]),
+                        None,
+                        Function(
+                            "arrayElement",
+                            [
+                                self.column("measurements_value"),
+                                lcp_index,
+                            ],
+                        ),
+                    ],
+                ),
+                self.column("transaction.duration"),
+            ],
+        )
+
+    def _resolve_apdex_function(self, args: Mapping[str, str], alias: str) -> SelectType:
+        if args["satisfaction"]:
+            return Function(
+                "apdex", [self.column("transaction.duration"), int(args["satisfaction"])], alias
+            )
+
+        return Function(
+            "apdex",
+            [
+                self._private_project_threshold_multi_function(),
+                Function("tupleElement", [self.resolve_field("project_threshold_config"), 2]),
+            ],
+            alias,
+        )
+
+    def _resolve_count_miserable_function(self, args: Mapping[str, str], alias: str) -> SelectType:
+        if args["satisfaction"]:
+            return Function(
+                "uniqIf",
+                [
+                    self.column(args["column"]),
+                    Function(
+                        "greater", [self.column("transaction.duration"), int(args["tolerated"])]
+                    ),
+                ],
+                alias,
+            )
+
+        return Function(
+            "uniqIf",
+            [
+                self.column(args["column"]),
+                Function(
+                    "greater",
+                    [
+                        self._private_project_threshold_multi_function(),
+                        Function(
+                            "multiply",
+                            [
+                                Function(
+                                    "tupleElement",
+                                    [self.resolve_field("project_threshold_config"), 2],
+                                ),
+                                4,
+                            ],
+                        ),
+                    ],
+                ),
+            ],
+            alias,
+        )
```

Comment:
For these functions, thoughts on having doing a single return that roughly has the structure of the snql_function and defining the bits as variables instead? eg. (lhs ain't the best variable name 🤷 but more this structure)
```python
    def _resolve_count_miserable_function(self, args: Mapping[str, str], alias: str) -> SelectType:
        if args["satisfaction"]:
            lhs = self.column("transaction.duration")
            rhs = int(args["tolerated"])
        else:
             lhs = self._private_project_threshold_multi_function()
             rhs = Function(
                 "multiply",
                 [
                     Function(
                         "tupleElement",
                         [self.resolve_field("project_threshold_config"), 2],
                     ),
                     4,
                 ],
            )

       return Function(
            "uniqIf",
            [
                self.column(args["column"]),
                Function(
                    "greater", [lhs, rhs]
                ),
            ],
            alias,
        )
```

Reply:
I just jammed everything into a variable called "function_args"

Reply:
changed my mind for _count_miserable, going to do the lhs/rhs you suggested